jetty-util/src/main/java/org/eclipse/jetty/util/Index.java

```diff
@@ -251,6 +333,21 @@ public Builder()
             this.contents = contents;
         }
 
+        /**
+         * A special purpose static builder for fast creation of specific Index type
+         * @param maxCapacity The max capacity of the index
+         * @param <V> The type of the index
+         * @return A case sensitive mutable Index tacking visible ASCII alphabet to a max capacity.
+         */
+        public static <V> Mutable<V> buildCaseSensitiveMutableVisibleAsciiAlphabet(int maxCapacity)
+        {
+            if (maxCapacity <= 0)
+                return EmptyTrie.instance(true);
+            if (maxCapacity <= ArrayTrie.MAX_CAPACITY)
```

Comment:
If `maxCapacity` is greater than `ArrayTrie.MAX_CAPACITY` you're going to return an empty trie. This breaks the principle of least surprise IMHO.

Reply:
ah yes... a TreeTrie it is then

Reply:
Oh, and I just remembered that a negative `maxCapacity` means unlimited. I think the whole method might need to be revisited.

Reply:
revisited

Reply:
Looks good now.